app/js/app.js

```diff
@@ -15,8 +15,21 @@ angular.module('angularMovieApp').config(function($stateProvider, $urlRouterProv
       templateUrl : 'partials/movie.html',
       controller  : 'movieController',
       resolve     : {
-        movie : function($stateParams, Movie) {
-          return Movie.fetchOne($stateParams.id);
+        movie: function($stateParams, Movie) {
+          return Movie.fetchCasting($stateParams.id)
```

Comment:
Ouch !
Nice pyramid of doom dude.
Did you know that [`$q.all`](https://docs.angularjs.org/api/ng/service/$q) supports hash of promises since Angular 1.1.x ? ;)